### PR TITLE
Fix nil exception to empty headers during metadata assignment

### PR DESCRIPTION
**Pull request by kaisecheng**
Fix: #141

This commit guard nil exception in metadata assignment

### CODE REVIEW

File: lib/logstash/inputs/kafka.rb [hunk 1]
```diff
@@ -374,6 +374,8 @@ def maybe_set_metadata(event, record)
     end
     if @metadata_mode.include?(:headers)
       record.headers.each do |header|
+        next if header.nil? || header.value.nil? || header.key.nil?
```
Comment (andsel) [inline lib/logstash/inputs/kafka.rb:377]: I tried to comment this line and ran the test, but was till green. Seems the test doesn't enter the loop.

**Review by andsel — CHANGES_REQUESTED** [review]
Tried locally and I think the test doesn't exercise the code, plus a minor suggestion to make it more declarative

File: lib/logstash/inputs/kafka.rb (hunk 1 above)
Comment (kaisecheng) [reply lib/logstash/inputs/kafka.rb]: Great that you spot it

File: lib/logstash/inputs/kafka.rb [hunk 2]
```diff
@@ -374,6 +374,8 @@ def maybe_set_metadata(event, record)
     end
     if @metadata_mode.include?(:headers)
       record.headers.each do |header|
```
[the remark this answers is not in the document]
Comment (kaisecheng) [reply lib/logstash/inputs/kafka.rb]: changed to `!header.nil? && !header.value.nil? && !header.key.nil?`

[the remark this answers is not in the document]
Comment (andsel) [reply lib/logstash/inputs/kafka.rb]: Good catch!

**Review by andsel — APPROVED** [review]
Once it's green, LGTM!

**Pull request merged**